ci-tools/host-runner/README.md

```diff
@@ -0,0 +1,255 @@
+```
+
+# Udev
+
+Udev rules are used to access USB and SD cards without root permissions. You can see this in `configuration.nix`. 
+
+If you get permission errors, you most likely need to add the device to the udev rules.
+
+Use the udev tools to figure out the `idVendor` and `idProduct` attributes of the device.
+
+```nix
+  services.udev.extraRules = ''
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6011", OWNER="${user}", GROUP="users"
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6001", OWNER="${user}", GROUP="users"
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="0424", ATTRS{idProduct}=="2640", OWNER="${user}", GROUP="users"
+    SUBSYSTEMS=="usb", ATTRS{idVendor}=="0424", ATTRS{idProduct}=="4050", OWNER="${user}", GROUP="users"
+    '';
+```
+
+# Download FPGA Image
+
+Download the latest FPGA image from [GitHub](https://github.com/chipsalliance/caliptra-sw/actions/workflows/fpga-image.yml). This job is scheduled to run once a week.
+
+This image should be copied to the host runner and stored at `$HOME/zcu104.img`.
+
+# Connect an FPGA to the host runner
+
+For an example, see `hostrunners/kir-0.nix` for 3 ZCU-104s managed with systemd. You should use Nix to manage these services so they are re-producible.
+
+## Generic instructions
+
+### RPI Setup
+
+1. Install Rust
+1. Install fpga-boss.
+    - `cargo install --git https://github.com/chipsalliance/caliptra-sw.git caliptra-fpga-boss --branch main`
+1. Install rtool
+    - `(cd ci-tools/github-runner/cmd/rtool && go build .)`
+1. Download the latest FPGA image from [GitHub](https://github.com/chipsalliance/caliptra-sw/actions/workflows/fpga-image.yml). This job is scheduled to run once a week.
+    - In the following examples, the image is stored in `$HOME/zcu-scripts/zcu104.img` for the Raspberry PI user.
+1. Get a GitHub Application private key for the CI. This is used to sign JWTs used by the FPGA runners.
+    - The template below stores this file at `/etc/secrets/caliptra-gce-ci-github-private-key-pem/prod`.
+
+### UDEV rules
+
+Add UDEV rules for the hardware that is involved to avoid running the CI as root.
+
+Here is an example from my RPI:
+
+```
+runner@caliptrarpi:~/zcu104-scripts $ cat /etc/udev/rules.d/99-fpga.rules
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6011", OWNER="runner", GROUP="runner"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="04e8", ATTRS{idProduct}=="6001", OWNER="runner", GROUP="runner"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="0424", ATTRS{idProduct}=="2640", OWNER="runner", GROUP="runner"
+SUBSYSTEMS=="usb", ATTRS{idVendor}=="0424", ATTRS{idProduct}=="4050", OWNER="runner", GROUP="runner"
+```
+
+These rules are for the FTDI / SDWIRE USB hosts, as well as the SD Cards that I use. You may have different vendors and products.
+
+## FPGA Bash Script Template
```

Comment:
A lot of these instructions are completed in the [FPGA image](https://github.com/chipsalliance/caliptra-sw/tree/main/ci-tools/fpga-image) right? If so, it would be good to mention that and have a link to it.

Reply:
Hmm that script is run on the FPGA itself - this would be on the hostrunner (RPI).

Reply:
Ahh okay, nevermind